dj_cqrs/_validation.py

```diff
@@ -0,0 +1,165 @@
+    if not transport_cls_location:
+        raise AssertionError('CQRS transport is not set.')
+
+    transport = import_string(transport_cls_location)
+    if not issubclass(transport, BaseTransport):
+        raise AssertionError(
+            'CQRS transport must be inherited from `dj_cqrs.transport.BaseTransport`.',
+        )
+
+
+def _validate_master(cqrs_settings):
+    default_master_settings = {
+        'master': {
+            'CQRS_AUTO_UPDATE_FIELDS': DEFAULT_MASTER_AUTO_UPDATE_FIELDS,
+            'CQRS_MESSAGE_TTL': DEFAULT_MASTER_MESSAGE_TTL,
+            'correlation_function': None,
+        },
+    }
+
+    if 'master' not in cqrs_settings:
+        cqrs_settings.update(default_master_settings)
+        return
+
+    master_settings = cqrs_settings['master']
+    assert isinstance(master_settings, dict), 'CQRS master configuration must be dict.'
+
+    if 'CQRS_AUTO_UPDATE_FIELDS' in master_settings:
+        assert isinstance(master_settings['CQRS_AUTO_UPDATE_FIELDS'], bool), (
+            'CQRS master CQRS_AUTO_UPDATE_FIELDS must be bool.'
+        )
+    else:
+        master_settings['CQRS_AUTO_UPDATE_FIELDS'] = DEFAULT_MASTER_AUTO_UPDATE_FIELDS
+
+    if 'CQRS_MESSAGE_TTL' in master_settings:
+        min_message_ttl = 1
+        message_ttl = master_settings['CQRS_MESSAGE_TTL']
+        if (message_ttl is not None) and (
+            not isinstance(message_ttl, int) or message_ttl < min_message_ttl
+        ):
+            # No error is raised for backward compatibility
+            # TODO: raise error in 2.0.0
+            logger.warning(
+                'Settings CQRS_MESSAGE_TTL=%s is invalid, using default %s.',
+                message_ttl, DEFAULT_MASTER_MESSAGE_TTL,
+            )
+            master_settings['CQRS_MESSAGE_TTL'] = DEFAULT_MASTER_MESSAGE_TTL
+    else:
+        master_settings['CQRS_MESSAGE_TTL'] = DEFAULT_MASTER_MESSAGE_TTL
+
+    correlation_func = master_settings.get('correlation_function')
+    if not correlation_func:
+        master_settings['correlation_function'] = None
+    elif not callable(correlation_func):
+        raise AssertionError('CQRS master correlation_function must be callable.')
+
+
+def _validate_replica(cqrs_settings):
+    queue = cqrs_settings.get('queue')
+    assert queue, 'CQRS queue is not set.'
+    assert isinstance(queue, str), 'CQRS queue must be string.'
+
+    default_replica_settings = {
+        'replica': {
+            'CQRS_MAX_RETRIES': DEFAULT_REPLICA_MAX_RETRIES,
+            'CQRS_RETRY_DELAY': DEFAULT_REPLICA_RETRY_DELAY,
+            'CQRS_DELAY_QUEUE_MAX_SIZE': DEFAULT_REPLICA_DELAY_QUEUE_MAX_SIZE,
+        },
+    }
+
+    if 'replica' not in cqrs_settings:
+        cqrs_settings.update(default_replica_settings)
+        return
+
+    replica_settings = cqrs_settings['replica']
+    assert isinstance(replica_settings, dict), 'CQRS replica configuration must be dict.'
+
+    if 'CQRS_MAX_RETRIES' in replica_settings:
```

Comment:
Could we move specific property validations to functions (for example _validate_cqrs_max_retries)?
For me It's hard to read when there is a complex logic at every property.

Reply:
Why not